### PR TITLE
MeOX & TMS GC-Derivatization Tool

### PR DESCRIPTION
**Pull request by wverastegui**
This PR implements a MeOX and TMS GC-derivatization tool that reads a list of SMILES from SMI and Tabular formats.

Closes https://github.com/RECETOX/galaxytools/issues/295

### CODE REVIEW

File: tools/gc_derivatization/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+owner: recetox
+remote_repository_url: "https://github.com/RECETOX/galaxytools/tree/master/tools/gc_derivatization"
+homepage_url: "https://github.com/RECETOX/galaxytools"
+categories:
+  - Metabolomics
+description: Performs GC derivatization.
+long_description: |
+    The GC-derivatization tool converts carbonyl groups to C═N–OCH3 (MeOX) and transforms acidic protons into -Si(CH3)3 (TMS). Key functionalities include checking for 
+    specific groups, removing derivatization groups, and adding derivatization groups to molecules.
+include:
```
Comment (bgruening) [inline tools/gc_derivatization/.shed.yml:12]: Try to get rid of include, its hard to maintain such a list.

File: tools/gc_derivatization/gc_derivatization.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                    <has_text text="CC=NOC"/>
+                    <has_text text="OC1=CC=CC=C1"/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help>
+        <![CDATA[
+        GC derivatization conducts in-silico derivatization using Methoximation (MeOX) and Trimethylsilylation (TMS).
+        Detailed documentation can be found in https://doi.org/10.1021/acs.analchem.7b01010.
```
Comment (bgruening) [inline tools/gc_derivatization/gc_derivatization.xml:37]:
```suggestion
```

File: tools/gc_derivatization/gc_derivatization.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<tool id="gc_derivatization" name="Gas chromatography derivatization" version="@TOOL_VERSION@+galaxy1" profile="21.09">
+    <description>This tool perform MeOX and TMS derivatization</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">gc-meox-tms</requirement>
+    </requirements>
```
Comment (bgruening) [inline tools/gc_derivatization/gc_derivatization.xml:8]: creator is not imported afaics

File: tools/gc_derivatization/.shed.yml (hunk 1 above)
Comment (wverastegui) [reply tools/gc_derivatization/.shed.yml]: Done [88ae07a](https://github.com/RECETOX/galaxytools/pull/503/commits/88ae07aff959d895e5c28b5ea2b11c7b30902d95)

File: tools/gc_derivatization/gc_derivatization.xml (hunk 2 above)
Comment (wverastegui) [reply tools/gc_derivatization/gc_derivatization.xml:37]: Edited [1515438](https://github.com/RECETOX/galaxytools/pull/503/commits/151543806594e8e6d01cce558b60a57bc6327ee3)

File: tools/gc_derivatization/gc_derivatization.xml (hunk 3 above)
Comment (wverastegui) [reply tools/gc_derivatization/gc_derivatization.xml:9]: Imported [1515438](https://github.com/RECETOX/galaxytools/pull/503/commits/151543806594e8e6d01cce558b60a57bc6327ee3)

File: tools/gc_derivatization/.shed.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+
+name: gc_derivatization
+owner: recetox
+remote_repository_url: "https://github.com/RECETOX/galaxytools/tree/master/tools/gc_derivatization"
+homepage_url: "https://github.com/RECETOX/galaxytools"
```
Comment (hechth) [inline tools/gc_derivatization/.shed.yml:5]:
```suggestion
homepage_url: "https://github.com/RECETOX/gc-meox-tms"
```

File: tools/gc_derivatization/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+categories:
+  - Metabolomics
+description: Performs GC derivatization.
```
Comment (hechth) [inline tools/gc_derivatization/.shed.yml:8]:
```suggestion
description: In silico derivatization for GC.
```

File: tools/gc_derivatization/gc_derivatization.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<tool id="gc_derivatization" name="Gas chromatography derivatization" version="@TOOL_VERSION@+galaxy1" profile="21.09">
```
Comment (hechth) [inline tools/gc_derivatization/gc_derivatization.xml:1]:
```suggestion
<tool id="gc_derivatization" name="GC derivatization" version="@TOOL_VERSION@+galaxy0" profile="21.09">
```

File: tools/gc_derivatization/gc_derivatization.xml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    <description>This tool perform MeOX and TMS derivatization</description>
```
Comment (hechth) [inline tools/gc_derivatization/gc_derivatization.xml:2]:
```suggestion
    <description>In silico MeOX and TMS derivatization tool</description>
```

File: tools/gc_derivatization/macros.xml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<macros>
+    <token name="@TOOL_VERSION@">1.0.0</token>
```
Comment (hechth) [inline tools/gc_derivatization/macros.xml:2]:
```suggestion
    <token name="@TOOL_VERSION@">1.0.1</token>
```

**Review by hechth — COMMENTED** [review]
Small changes so far - the versions need to be fixed, I will test the package locally and see what happens.

**Pull request merged**